### PR TITLE
Function for reference intensity

### PR DESCRIPTION
**Pull request by KonanAllaly**
Hi Jozef, 

I closed the previous pull request because I did a wrong manipulation and some functions for the 3D implementation (I am working on) got updated into the master branch. So I cleaned the branch, so the changes you will see in this pull request are just related to the reference intensity calculation. Thank you

### CODE REVIEW

File: ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c [hunk 1]
```diff
@@ -75,25 +73,31 @@ bool lagrangeanExplicit2DCurveSegmentation(Image_Data2D inputImage2D, const Lagr
...
 
     //get edge detector
-    
     for (size_t i = 0; i < dataDimension; i++)
     {
-        edge_detector[i] = edgeDetector(abs_val_grad[i], edge_detector_coef);
+        edge_detector[i] = edgeDetector(abs_val_grad[i], pSegmentationParams->edgeCoef);
     }
 
+    unsigned char saveEdge[] = "C:/Users/Konan Allaly/Documents/Tests/Curves/Output/edge_detector.raw";
```
Comment (JozoUrban) [inline ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c]: please remove or at least comment out the absolute path

File: ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c [hunk 2]
```diff
@@ -75,25 +73,31 @@ bool lagrangeanExplicit2DCurveSegmentation(Image_Data2D inputImage2D, const Lagr
...
     if (!pSegmentationParams->open_curve)
     {
         for (size_t i = 0; i < dataDimension; i++)
         {
-            similar_intensity_detector[i] = similarIntensityDetector(inputImage2D.imageDataPtr[i], ref_intensity, similar_intensity_detector_coef);
+            similar_intensity_detector[i] = similarIntensityDetector(inputImage2D.imageDataPtr[i], pSegmentationParams->reference_intensity, pSegmentationParams->intensityCoef);
         }
     }
 
+    unsigned char saveG2[] = "C:/Users/Konan Allaly/Documents/Tests/Curves/Output/similarIntensityDetect.raw";
```
Comment (JozoUrban) [inline ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c]: please remove or at least comment out the absolute path

File: ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c [hunk 3]
```diff
@@ -274,15 +276,16 @@ bool lagrangeanSemiImplicit2DCurveSegmentation(Image_Data2D inputImage2D, const
...
+        ptmp[i] = edgeDetector(ptmp[i], pSegmentationParams->edgeCoef);
     }
 
     Image_Data2D edge = { inputImage2D.height, inputImage2D.width, ptmp };
 
     Point2D centroid = getCurveCentroid(pSegmentationParams->pinitial_condition);
     size_t centroid_i = (size_t)(centroid.y + 0.5);
     size_t centroid_j = (size_t)(centroid.x + 0.5);
-    dataType ref_intensity = inputImage2D.imageDataPtr[x_new(centroid_j, centroid_i, inputImage2D.width)];
+    //The reference intensity is now computed in the main and incorporated to the segmentation parameters
```
Comment (JozoUrban) [inline ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c]: this comment can be removed

**Review by JozoUrban — CHANGES_REQUESTED** [review]

File: ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c (hunk 1 above)
Comment (KonanAllaly) [reply ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c]: Done

File: ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c (hunk 2 above)
Comment (KonanAllaly) [reply ImageInLib/ImageInLib/src/segmentation2D_lagrangean.c]: done

**Review by JozoUrban — APPROVED** [review]

**Pull request merged**